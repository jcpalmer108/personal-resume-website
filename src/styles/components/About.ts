import styled from "styled-components";
import {
  desktopBreakpoint,
  redstone,
  seashell_peach,
  tabletBreakpoint,
} from "../variables";

export const AboutWrapper = styled.div`
  display: flex;
  flex-direction: column;

  @media (min-width: ${tabletBreakpoint}px) {
    flex-direction: row;
  }
`;

export const Content = styled.div`
  padding-bottom: 40px;

  @media (min-width: ${tabletBreakpoint}px) {
    padding-bottom: 0px;
  }
`;

export const Title = styled.h2`
  padding-bottom: 20px;
`;

export const Description = styled.p`
  margin: 0px;
  padding-bottom: 20px;

  &:last-child {
    padding-bottom: 0px;
  }
`;

export const Schooling = styled.div`
  display: flex;
  flex-direction: column;
  padding-bottom: 15px;

  @media (min-width: ${tabletBreakpoint}px) {
    min-width: fit-content;
    padding-left: 70px;
    padding-bottom: 0px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    min-width: 290px;
    padding-left: 125px;
  }
`;

export const School = styled.div`
  padding-top: 25px;
  display: flex;
  flex-direction: column;
  align-items: left;
  justify-content: center;

  @media (min-width: ${tabletBreakpoint}px) {
    padding-top: 35px;
  }
`;

export const Degree = styled.span`
  padding-top: 12px;

  @media (min-width: ${tabletBreakpoint}px) {
    font-size: 14px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    font-size: 16px;
    padding-top: 10px;
  }
`;

export const Action = styled.div`
  display: flex;
  flex-direction: column;
  padding: 0px 15px;

  @media (min-width: ${tabletBreakpoint}px) {
    flex-direction: row;
    padding: 0px;
  }
`;

export const Connect = styled.div`
  flex-grow: 2;
  background-color: ${seashell_peach};
  display: flex;
  height: 100px;
  flex-direction: column;
  padding: 25px 15px;

  @media (min-width: ${tabletBreakpoint}px) {
    height: 70px;
    padding: 40px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    height: 75px;
    padding: 50px;
  }
`;

export const SocialLabel = styled.span`
  font-size: 20px;
  font-family: Butler-Regular;
  padding-bottom: 26px;

  @media (min-width: ${tabletBreakpoint}px) {
    font-size: 30px;
    padding-bottom: 20px;
  }
`;

export const Socials = styled.div`
  display: flex;
  height: 100%;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  row-gap: 23px;

  @media (min-width: ${tabletBreakpoint}px) {
    justify-content: space-between;
    height: 20px;
  }
`;

export const Link = styled.a`
  text-decoration: none;
  color: black;
  display: flex;
  flex-direction: row;
  font-size: 16px;
  align-items: center;
  justify-content: left;
  column-gap: 10px;
  width: 33%;
  font-family: Gilroy-Medium;

  @media (min-width: ${tabletBreakpoint}px) {
    width: fit-content;
    height: fit-content;
    justify-content: center;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    width: 20%;
    justify-content: left;
  }
`;

export const Icon = styled.img`
  height: 16px;
  width: 16px;

  @media (min-width: ${tabletBreakpoint}px) {
    height: 19px;
    width: 19px;
  }

  @media (min-width: ${tabletBreakpoint}px) {
    height: 22px;
    width: 22px;
  }
`;

export const Label = styled.span`
  font-size: 14px;

  @media (min-width: ${tabletBreakpoint}px) {
    font-size: 16px;
    padding-top: 4px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    font-size: 19px;
    padding-top: 4px;
  }
`;

export const Prompt = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: white;
  height: 60px;
  background: ${redstone};
  padding: 20px;
  text-align: center;

  @media (min-width: ${tabletBreakpoint}px) {
    width: 130px;
    height: 90px;
    padding: 30px;
  }

  @media (min-width: ${desktopBreakpoint}px) {
    width: 148px;
    height: 90px;
    padding: 43px;
  }
`;

export const Question = styled.span`
  font-size: 16px;
  font-family: Butler-Medium;

  @media (min-width: ${tabletBreakpoint}px) {
    font-size: 18px;
  }
`;

export const StartProject = styled.a`
  text-decoration: none;
  font-size: 12px;
  font-family: Gilroy-Medium;
  padding-top: 23px;
  padding-bottom: 7px;
  color: white;
  border-bottom: 1px solid white;
`;

export const BottomSpacing = styled.div`
  height: 0px;

  @media (min-width: ${desktopBreakpoint}px) {
    height: 80px;
  }
`;
