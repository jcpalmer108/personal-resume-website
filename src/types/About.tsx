import { Sections } from "./content";

export type AboutProps = {
  content: Sections | undefined;
};
