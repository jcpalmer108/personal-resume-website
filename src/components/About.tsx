import Section from "./Section"
import { AboutProps } from "../types/About"
import { 
  AboutWrapper, 
  Action, 
  Content, 
  Schooling, 
  School, 
  Link, 
  Icon, 
  Label,
  Connect,
  Prompt,
  Degree, 
  Socials,
  SocialLabel,
  StartProject,
  Question,
  Description,
  Title,
  BottomSpacing
} from "../styles/components/About"
import { Education } from "../types/content"
import { SocialLink } from "../types/socials"
import socials from "../assets/content/socials.json"

export default function About({ content }: AboutProps) {
  const { links } = socials;

  if(content === undefined) {
    return null;
  }

  return (
    <div id="about" data-testid="About">
      <Section label={content?.label} >
        <Title data-testid="Title">{content?.title}</Title>
        <AboutWrapper data-testid="Content">
          <Content>
            {content?.description && content.description.map((paragraph, index) => <Description data-testid={`Description ${index + 1}`} key={`Description ${index + 1}`}>{paragraph}</Description>)}
          </Content>
          <Schooling data-testid="Schooling">
            <h3>Education</h3>
            {content?.subSection && content.subSection.education?.map((item: Education, index) => 
              <School data-testid={`School ${index + 1}`} key={`School ${index + 1}`}>
                <h4>{item.school}</h4>
                <Degree>{item.degree} - {item.timeline.start}-{item.timeline.end}</Degree>
              </School>
            )}
          </Schooling>
        </AboutWrapper>
      </Section>
      <Section wide>
        <Action>
          <Connect data-testid="Connect">
            <SocialLabel>Connect with Me</SocialLabel>
            <Socials>
              {links && links.map(({key, label, url}: SocialLink) => 
                <Link href={url} key={key}>
                  <Icon src={require("../assets/images/socials/" + key + ".svg")} />
                  <Label>{label}</Label>
                </Link>
              )}
            </Socials>
          </Connect>
          <Prompt>
            <Question>What are you waiting for?</Question>
            <StartProject data-testid="Start a Project" href="#contact">START A PROJECT</StartProject>
          </Prompt>
        </Action>
      </Section>
      <Section wide bottom>
        <BottomSpacing />
      </Section>
    </div>
  )
}