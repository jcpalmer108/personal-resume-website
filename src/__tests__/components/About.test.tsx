import { render, screen } from '@testing-library/react';
import 'jest-styled-components'
import About from '../../components/About';

jest.mock('../../components/Button')

const params = {
  key: "about",
  label: "About",
  inMenu: true,
  title: "test title",
  description: [
    "test contents 1",
    "test contents 2"
  ],
  subSection: {
    education: [
      {
        key: "webster",
        school: "Webster University",
        degree: "Bachelors of Science in Computer Science",
        timeline: {
          start: "2014",
          end: "2017"
        }
      },
      {
        key: "harvard",
        school: "Harvard University",
        degree: "Bachelors of Smartiness in Computer Science",
        timeline: {
          start: "2027",
          end: "2030"
        }
      }
    ]
  }
}

const mockSocials = [
  {
    key: "github",
    label: "Github",
    url: "https://www.github.com/jcpalmer108/"
  },
  {
    key: "leetcode",
    label: "Leetcode",
    url: "https://leetcode.com/darthToaster/"
  }
]

describe('About', () => {
  test('renders if only required params are passed in', () => {
    render(
      <About content={params} />
    )
  
    expect(screen.getByTestId('About')).toMatchSnapshot()
    expect(screen.getByTestId('Title')).toHaveTextContent(params.title)

    params.description.forEach((item, index) => {
      expect(screen.getByTestId(`Description ${index + 1}`)).toHaveTextContent(item)
    })
    
    expect(screen.getByTestId('Schooling').childNodes).toHaveLength(params.subSection.education.length + 1)
    params.subSection.education.forEach((item, index) => {
      expect(screen.getByTestId(`School ${index + 1}`).childNodes[0]).toHaveTextContent(item.school)
      expect(screen.getByTestId(`School ${index + 1}`).childNodes[1]).toHaveTextContent(item.degree + ' - ' + item.timeline.start + '-' + item.timeline.end)
    })

    expect(screen.getByTestId('Connect')).toMatchSnapshot()
    mockSocials.forEach((link) => {
      expect(screen.getByTestId('Connect')).toHaveTextContent(link.label)
      expect(screen.getByRole('link', { name: link.label })).toBeTruthy()
      expect(screen.getByRole('link', { name: link.label })).toHaveAttribute(
        'href',
        link.url
      )
    })

    expect(screen.getByTestId('Start a Project')).toMatchSnapshot()
    expect(screen.getByRole('link', { name: 'START A PROJECT' })).toHaveAttribute(
      'href',
      '#contact'
    )
  })
})
