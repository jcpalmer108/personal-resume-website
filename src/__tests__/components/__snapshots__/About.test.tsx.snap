// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders if only required params are passed in 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  padding-bottom: 40px;
}

.c2 {
  margin: 0px;
  padding-bottom: 20px;
}

.c2:last-child {
  padding-bottom: 0px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 15px;
}

.c4 {
  padding-top: 25px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  padding-top: 12px;
}

@media (min-width:1155px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:720px) {
  .c1 {
    padding-bottom: 0px;
  }
}

@media (min-width:720px) {
  .c3 {
    min-width: -webkit-fit-content;
    min-width: -moz-fit-content;
    min-width: fit-content;
    padding-left: 70px;
    padding-bottom: 0px;
  }
}

@media (min-width:1024px) {
  .c3 {
    min-width: 290px;
    padding-left: 60px;
  }
}

@media (min-width:720px) {
  .c4 {
    padding-top: 35px;
  }
}

@media (min-width:720px) {
  .c5 {
    font-size: 14px;
  }
}

@media (min-width:1024px) {
  .c5 {
    font-size: 16px;
    padding-top: 10px;
  }
}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

<div
  class="c0"
  data-testid="About"
>
  <div
    class="c1"
  >
    <p
      class="c2"
      data-testid="Description 1"
    >
      test contents 1
    </p>
    <p
      class="c2"
      data-testid="Description 2"
    >
      test contents 2
    </p>
  </div>
  <div
    class="c3"
    data-testid="Schooling"
  >
    <h3>
      Education
    </h3>
    <div
      class="c4"
      data-testid="School 1"
    >
      <h4>
        Webster University
      </h4>
      <span
        class="c5"
      >
        Bachelors of Science in Computer Science
         - 
        2014
        -
        2017
      </span>
    </div>
    <div
      class="c4"
      data-testid="School 2"
    >
      <h4>
        Harvard University
      </h4>
      <span
        class="c5"
      >
        Bachelors of Smartiness in Computer Science
         - 
        2027
        -
        2030
      </span>
    </div>
  </div>
</div>
`;

exports[`About renders if only required params are passed in 2`] = `
.c0 {
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  background-color: #FFF7F2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 25px 15px;
}

.c1 {
  font-size: 20px;
  font-family: Butler-Regular;
  padding-bottom: 26px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  row-gap: 23px;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-column-gap: 10px;
  column-gap: 10px;
  width: 33%;
  font-family: Gilroy-Medium;
}

.c4 {
  height: 16px;
  width: 16px;
}

.c5 {
  font-size: 14px;
}

@media (min-width:1155px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {
  .c0 {
    height: 70px;
    padding: 40px;
  }
}

@media (min-width:1024px) {
  .c0 {
    height: 75px;
    padding: 50px;
  }
}

@media (min-width:720px) {
  .c1 {
    font-size: 30px;
    padding-bottom: 20px;
  }
}

@media (min-width:720px) {
  .c2 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    height: 20px;
  }
}

@media (min-width:720px) {
  .c3 {
    width: -webkit-fit-content;
    width: -moz-fit-content;
    width: fit-content;
    height: -webkit-fit-content;
    height: -moz-fit-content;
    height: fit-content;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (min-width:1024px) {
  .c3 {
    width: 20%;
    -webkit-box-pack: left;
    -webkit-justify-content: left;
    -ms-flex-pack: left;
    justify-content: left;
  }
}

@media (min-width:720px) {
  .c4 {
    height: 19px;
    width: 19px;
  }
}

@media (min-width:720px) {
  .c4 {
    height: 22px;
    width: 22px;
  }
}

@media (min-width:720px) {
  .c5 {
    font-size: 16px;
    padding-top: 4px;
  }
}

@media (min-width:1024px) {
  .c5 {
    font-size: 19px;
    padding-top: 4px;
  }
}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

<div
  class="c0"
  data-testid="Connect"
>
  <span
    class="c1"
  >
    Connect with Me
  </span>
  <div
    class="c2"
  >
    <a
      class="c3"
      href="https://www.github.com/jcpalmer108/"
    >
      <img
        class="c4"
        src="[object Object]"
      />
      <span
        class="c5"
      >
        Github
      </span>
    </a>
    <a
      class="c3"
      href="https://leetcode.com/darthToaster/"
    >
      <img
        class="c4"
        src="[object Object]"
      />
      <span
        class="c5"
      >
        Leetcode
      </span>
    </a>
    <a
      class="c3"
      href="https://www.linkedin.com/in/jenna-palmer/"
    >
      <img
        class="c4"
        src="[object Object]"
      />
      <span
        class="c5"
      >
        LinkedIn
      </span>
    </a>
    <a
      class="c3"
      href="https://twitter.com/jpalmerproject/"
    >
      <img
        class="c4"
        src="[object Object]"
      />
      <span
        class="c5"
      >
        Twitter
      </span>
    </a>
    <a
      class="c3"
      href="mailto:jenna@jennapalmer.info"
    >
      <img
        class="c4"
        src="[object Object]"
      />
      <span
        class="c5"
      >
        Email
      </span>
    </a>
  </div>
</div>
`;

exports[`About renders if only required params are passed in 3`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 12px;
  font-family: Gilroy-Medium;
  padding-top: 23px;
  padding-bottom: 7px;
  color: white;
  border-bottom: 1px solid white;
}

@media (min-width:1155px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

@media (min-width:1024px) {

}

@media (min-width:720px) {

}

<a
  class="c0"
  data-testid="Start a Project"
  href="#contact"
>
  START A PROJECT
</a>
`;
